muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.muzzle;
+
+import static java.util.Collections.emptyMap;
+
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.muzzle.ClassRef;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Map;
+
+/**
+ * This class is used to access {@code getMuzzleReferences} method from the {@link
+ * InstrumentationModule} class. That method is not visible in the source code and instead is
+ * generated automatically during compilation by the {@code
+ * io.opentelemetry.instrumentation.javaagent-codegen} Gradle plugin.
+ */
+class MuzzleReferencesAccessor {
```

Comment:
Perhaps `javaagent-codegen` should add an interface that has `getMuzzleReferences` to instrumentation module so you wouldn't need to use reflection.

Reply:
I assume that MethodHandle has no performance overhead on invocation. But I will think about this idea, it may be more elegant indeed.

Reply:
I will try this idea as a follow-up optimisation. I like it, but prefer to finish current iteration of removing ClassRef and friends from public API.

Reply:
This will also be easier to extend with other muzzle-generated methods.

Reply:
If it is possible to build extension jars that contain instrumentation modules then all of the muzzle and codegen stuff has become part of our api. Have you thought about how to version it and how to handle cases where agent is incompatible with codegen version used?

Reply:
Ohhh, good question... The only vague though I have atm is "javagent must support all previous stable versions of codegen within any given major version." Which assumes "in sync" versioning of the agent and gradle plugins.

Reply:
> If it is possible to build extension jars that contain instrumentation modules then all of the muzzle and codegen stuff has become part of our api

oh no, this is a great point, so maybe we shouldn't get rid of these classes from our public API? 😭

Reply:
These classes should be part of the contract between muzzle generation and muzzle verification. But no end-user, including extension authors, should see them.